src/detect-filestore.c

```diff
@@ -466,6 +471,7 @@ static int DetectFilestoreSetup (DetectEngineCtx *de_ctx, Signature *s, const ch
     }
 
     s->flags |= SIG_FLAG_FILESTORE;
+    de_ctx->filestore_cnt++;
```

Comment:
Having trouble understanding the introduction of this if the goal is to have a limit per sgh.
Questions:
1. Why are we trying to store `filestore_cnt` per `de_ctx` if the aim is to keep the limit on sgh?
2. Could the `de_ctx` limit in your patch not hit much sooner than the sgh limit in a worst case scenario?

Reply:
There are not more signatures in a sgh than are existing in the whole detection engine context

> Could the de_ctx limit in your patch not hit much sooner than the sgh limit in a worst case scenario?

Indeed

Reply:
> There are not more signatures in a sgh than are existing in the whole detection engine context

Agreed. Should we then update the commit message and the ticket to say that the limit is for detection engine and not per sgh?

Reply:
I updated the commit message in https://github.com/OISF/suricata/pull/10297

The limit is enforced on detection engine to have it per sgh as well.